### ISSUE TITLE
Error message not helpfull

### ISSUE THREAD

**Issue by janwilmans** [docopt/docopt.cpp#82]
>phooey.exe -h
Docopt usage string could not be parsed
Mismatched '['

message has no linenr, nor any text/tokens of line offending line.
I will add the reproduction scenario as soon as I figure out what line it is, by binary search elimination.

**Comment by janwilmans** [comment]
found it;
```
static const char USAGE[] =
R"(DebugviewConsole )" VERSION_STR
R"(
    Usage: 
        DebugviewConsole [-a (value)] [-i <pattern>, --include <pattern>]... [-e <pattern>, --exclude <pattern>]... 

    Options:
      -h --help             Show this screen
      -a value              auto-newline [default: true]  // offending line
      -i <pattern>, --include <pattern>     include filter, may be specified multiple times
      -e <pattern>, --exclude <pattern>     exclude filter, may be specified multiple times

)";
```

**Comment by jaredgrubb** [comment]
Yes, the error messages are not helpful. This is a limitation that we inherit from the docopt sources. I'd like to see this fixed, but I think it's a more global problem that just the C++ implementation. I'm always open to suggestions on this.

**Comment by janwilmans** [comment]
Ok, so what is efficient, should I try to improve this in the python implementation of docopt and you can pull it in / convert that somehow?

I guess better error message would require more script/detailed parsing or analysis of the parser output ?

**Comment by Kerndog73** [comment]
It seems like almost all errors fall though to [this point](https://github.com/docopt/docopt.cpp/blob/69682578d4dbb59b297a15ea537f2b1b8735c7bc/docopt.cpp#L655). Simply printing out all the arguments that the user passed isn't useful. That line might as well be `throw DocoptArgumentError("Error");`. Though, to be fair, the python implementation has the same problem so maybe I'm complaining in the wrong place!

It seems like fixing this would involve rewriting a lot of code. I really like this library and I'm _almost_ willing to do that myself but I don't think it would be worth the effort (plus I'm pretty lazy!).

**Comment by jaredgrubb** [comment]
I would love to see better error messages but I think it would be _a lot_ of work -- almost a full rewrite. I would be open to it, but I think it's a big effort. This repo was mostly a port of the Python one, which (as you note) suffers from the same problem. 

If someone wants to start that, I would be willing to work with them to do it in a way that is going to be successful.
